### PR TITLE
Fix items section description height

### PR DESCRIPTION
**Pull request by Bars92**
- change ready reports service to wait for ready-doc element for better stability
- fixed items section description is not considered in dynamic height
- added description as markdown
- modified MarkdownSection to react component to enable ref

image

### CODE REVIEW

**Comment by agamm** [comment]
@Bars92 do the 2 last points regarding markdown affect the sane json? (eg the sane-doc-reports too?)

**Comment by Bars92** [comment]
@agamm good point. Lets make sure items section description is markdown supported in docs as well since it should be (missed it here)

File: src/components/Sections/SectionMarkdown.js [hunk 1]
```diff
@@ -23,176 +23,178 @@ import ins from 'markdown-it-ins';
...
+        break;
+      case 'hr':
+        res = (<hr {...props} />);
+        break;
+      case 'mdbtn': {
+        res = SectionMarkdown.createBtn(props, children);
+        break;
       }
+      case 'img': {
+        const srcArr = props.src.split('=size=');
```
Comment (agamm) [inline src/components/Sections/SectionMarkdown.js:62]: what is going on here? ('=size='?)

**Comment by agamm** [comment]
> @agamm good point. Lets make sure items section description is markdown supported in docs as well since it should be (missed it here)

Do you have markdown examples of that?

File: src/components/Sections/SectionMarkdown.js (hunk 1 above)
Comment (Bars92) [reply src/components/Sections/SectionMarkdown.js:62]: existing option to resize image in markdown similar to https://gist.github.com/uupaa/f77d2bcf4dc7a294d109

**Comment by Bars92** [comment]
@agamm yes you can take src/components/Layouts/ReportLayout.js

**Comment by Bars92** [comment]
@dorsha can we please review this? this is getting higher priority

File: src/components/Sections/SectionMarkdown.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable react/prop-types */
 import './SectionMarkdown.less';
 import '../../../assets/styles/railscasts.css';
-import React from 'react';
+import React, { Component } from 'react';
```
Comment (dorsha) [inline src/components/Sections/SectionMarkdown.js:4]: There are tons of changes in this file, there is no good way to review this.
What was changed here?

Comment (Bars92) [reply src/components/Sections/SectionMarkdown.js:4]: as i mentioned `modified MarkdownSection to react component to enable ref`, nothing else.

File: reportService/reportsServer.js [hunk 3]
```diff
@@ -130,7 +130,9 @@ const PAGE_MARGIN = 60;
     await page.goto('file://' + baseUrl + '/' + tmpReportName, {waitUntil: 'networkidle0'});
     await page.emulateMedia('screen');
     await page._client.send('Emulation.clearDeviceMetricsOverride');
-    await page.waitFor(5000); // wait for animations
+    await page.waitForSelector('#ready-doc', {
+      timeout: 10000
```
[the remark this answers is not in the document]
Comment (dorsha) [reply reportService/reportsServer.js:134]: Isn't 10 seconds is too long?

**Pull request merged**